mantidproject/mantid
# Apply changes from clang-tidy: modernize-use-default. (Part 1)

Description of work.

This uses clang-tidy's modernize-use-default tool to replace default bodies of special member functions with `= default;`. I then did quite a bit of cleanup, such as remove default constructors and destructors that would be automatically generated anyway.

**To test:**



code review: If there are missing symbols, the build should fail. I'm not sure how great our test coverage of constructors is. There are a number of classes with private virtual destructors. Double-check I didn't delete any of those, potentially making the destructor public. 

There is no GitHub issue associated with this PR.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
